File: Client/Branch/Project/Assets/Scripts/UI/UILoginPanel.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class UILoginPanel : UIBase
{
    private Button _Btn_EditorLogin;
    public Button Btn_EditorLogin
    {
        get
        {
            return Get<Button>("Widget/Editor/Btn_EditorLogin");
        }
    }
    public void Start()
    {
        if (Btn_EditorLogin != null)
        {
            Btn_EditorLogin.onClick.AddListener(OnClickBtn);
        }
    }

    public void OnClickBtn()
    {
        CSSceneManager.Instance.Load(ESceneType.GameScene, null, null, OnLoadedGameScene);
    }

    private void OnLoadedGameScene()
    {
        UIManager.Instance.ClosePanel<UILoginPanel>();
        UIManager.Instance.CreatePanel<UIMainMenuPanel>();
    }
}

File: Client/Branch/Project/luaRes/ui/panels/UILoginPanel.lua
---登入界面
---@class UILoginPanel:UIBase
local UILoginPanel = {}

--region 组件获取
function UILoginPanel:GetBtn_EditorLogin()
    if UILoginPanel.mGetBtn_EditorLogin == nil then
        UILoginPanel.mGetBtn_EditorLogin = UILoginPanel:GetCurComp("Widget/Editor/Btn_EditorLogin", "Button")
    end
    return UILoginPanel.mGetBtn_EditorLogin
end
--endregion

function UILoginPanel:Init()
    if (self:GetBtn_EditorLogin()) then
        self:GetBtn_EditorLogin().onClick:AddListener(UILoginPanel.OnClickBtn_EditorLogin);
    end
end

function UILoginPanel.OnClickBtn_EditorLogin()
    CS.CSSceneManager.Instance:Load(CS.ESceneType.GameScene, nil, nil, UILoginPanel.OnLoadedGameScene);
end

function UILoginPanel.OnLoadedGameScene()
    uimanager:ClosePanel("UILoginPanel")
    uimanager:CreatePanel("UIMainMenusPanel")
end

function UILoginPanel:Show()
end

function ondestroy()
end

return UILoginPanel